### PR TITLE
[DKP 2.2] Licensing Update

### PR DESCRIPTION
**Pull request by mwrightd2**
## Jira Ticket

https://jira.d2iq.com/browse/D2IQ-86399

## Description of changes being made

Two new Enterprise and Essential licensing topics to fit into the Kommander licensing section for 2.2

### Preview

You can preview the docs build using the following URL, adding the PR # where specified:
http://docs-d2iq-com-pr-4226.s3-website-us-west-2.amazonaws.com/

## Checklist

- [ ] Test all commands and procedures, if applicable.
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/blob/main/CONTRIBUTING.md) for more information.

### CODE REVIEW

**Comment by mwrightd2** [comment]
> DKP ENTERPRISE
> 
> Under this header, the linked object appears to be missing: http://docs-d2iq-com-pr-4226.s3-website-us-west-2.amazonaws.com/dkp/kommander/2.2/licensing/enterprise/#dkp-enterprise-overview
> 
> http://docs-d2iq-com-pr-4226.s3-website-us-west-2.amazonaws.com/dkp/kommander/2.2/licensing/enterprise/#compatible-infrastructure Recommend we also put in "other pre provisioned virtual infrastructure" or something to that accord underneath VMware vSphere
> 
> http://docs-d2iq-com-pr-4226.s3-website-us-west-2.amazonaws.com/dkp/kommander/2.2/licensing/enterprise/#catalog-applications Do we plan to include Cassandra and Jenkins as catalog applications for this release? I thought they were pushed to 2.3 or later.
> 
> http://docs-d2iq-com-pr-4226.s3-website-us-west-2.amazonaws.com/dkp/kommander/2.2/licensing/enterprise/#faq Will their old license still work with DKP 2.2? Or will they need to get a new license from their account team to ensure that things continue to work?
> 
> DKP ESSENTIAL
> 
> Is the name of the product "DKP Essential (singular)" or "DKP Essentials (plural)". My understanding was that it was "Essentials". But all of the copy refers to "Essential"
> 
> http://docs-d2iq-com-pr-4226.s3-website-us-west-2.amazonaws.com/dkp/kommander/2.2/licensing/essential/#dkp-essential-overview below linked graphic appears to be missing.
> 
> http://docs-d2iq-com-pr-4226.s3-website-us-west-2.amazonaws.com/dkp/kommander/2.2/licensing/essential/#compatible-infrastructure Recommend we also put in "other pre provisioned virtual infrastructure" or something to that accord underneath VMware vSphere
> 
> Otherwise, looks good to me.

Thanks for the review @dmennell! I'll make changes where concrete and wait for feedback from other reviewers on the others.

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+* Bare metal servers
+* VMware vSphere   
+
+Cloud deployment options for DKP Enterprise include:
+
+* Amazon Web Services (AWS)
+* Amazon Elastic Kubernetes Service (EKS)
+* Azure Kubernetes Service (AKS)
+* Google Kubernetes Engine (GKE)
+* Lightweight Kubernetes (K3s)
```
Comment (cmastr) [inline pages/dkp/kommander/2.2/licensing/enterprise/index.md:31]: I'm pretty sure we don't support K3s. Where did that come from?

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+## Compatible infrastructure
+
+DKP Enterprise operates across D2iQ's entire range of cloud, on-premise, edge, and air-gapped infrastructures and has support for various OSs, including immutable OSs.
+
+Deployment options for DKP Enterprise include:
+
+* Bare metal servers
+* VMware vSphere   
+
+Cloud deployment options for DKP Enterprise include:
```
Comment (cmastr) [inline pages/dkp/kommander/2.2/licensing/enterprise/index.md:25]: I'm thinking we should NOT list these here and just point to the OS table that already talks about all the combinations we officially support:

https://docs.d2iq.com/dkp/konvoy/2.1/supported-operating-systems/

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+---
+layout: layout.pug
+navigationTitle: DKP Enterprise
+title: DKP Enterprise
```
Comment (cmastr) [inline pages/dkp/kommander/2.2/licensing/enterprise/index.md:4]: This looks really good!! I love the pictures. When we get into reorganization for Enterprise and Essential, we should use these images in the introduction to DKP section (to be created).

File: pages/dkp/kommander/2.2/licensing/essential/index.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+Deployment options for DKP Essential include:
+
+* Bare metal servers
+* VMware vSphere   
+
+Cloud deployment options for DKP Essential include:
+
+* Amazon Web Services (AWS)
+* Microsoft Azure
+* Google Cloud Platform (GCP)
```
Comment (cneth) [inline pages/dkp/kommander/2.2/licensing/essential/index.md:29]: See comments above.    GCP is not supported.

File: pages/dkp/kommander/2.2/licensing/essential/index.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+For the basics on standing up a DKP Essential cluster in one of the previously listed environments of your choice, see [Choose Infrastructure][choose-infrastructure].  
+
+<p class="message--note"><strong> NOTE:</strong> Infrastructure options are dependent upon license type. Some infrastructure deployment options listed within the corresponding link above may not be available for DKP Essential users. For an expanded list of supported infrastructure, see <a href="https://docs.d2iq.com/dkp/kommander/2.2/licensing/enterprise/">DKP Enterprise</a>.</p>
```
Comment (cneth) [inline pages/dkp/kommander/2.2/licensing/essential/index.md:33]: I am not aware of any direct tie between licensing and what infra providers are supported.    

True, you cannot install Essential (or even Enterprise) on AKS/EKS, whereas Enterprise can provide lifecycle management for managed clusters based on those platforms, but I think this is a confusing way to word that.

@GoelDeepak  - is this really how we want to explain this?

File: pages/dkp/kommander/2.2/licensing/essential/index.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+## FAQ
+
+* I'm getting a denial message when I attempt to attach or manage clusters with my DKP Essential self-managed cluster, why?
+
+DKP Essential only provides for the ability to stand up an individual, self-managed kubernetes cluster. Attaching or managing other clusters via a management cluster is only available by upgrading to [DKP Enterprise][dkpenterprise]
+
+* I used to be able to work with one free attached cluster, and now I can't see it in the UI, why?
+
+Prior to DKP version 2.2 you could create or attach 1 additional cluster without a license. If you have a 2.1 cluster that utilized that 1 additional cluster and upgrade to 2.2, the upgrade will automatically put you into the DKP Essential UI, and you will no longer have access to that additional cluster in the UI. However, the additional cluster will still appear within the CLI. To resolve this, enter your [DKP Enterprise][dkpenterprise] license in either the UI or CLI, and the previously attached cluster and deployed applications will reappear within the UI.
```
Comment (cneth) [inline pages/dkp/kommander/2.2/licensing/essential/index.md:75]: I think we should probably have a meeting about how to message this.  My understanding is that this issue is going to be discussed with the field teams next week, I think we should hold off until that meeting is over to understand how best to message this.

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+* Google Kubernetes Engine (GKE)
+* Lightweight Kubernetes (K3s)
+* Microsoft Azure
+* Google Cloud Platform (GCP)
+
+For the basics on standing up a DKP Enterprise cluster in one of the previously listed environments of your choice, see [Choose Infrastructure][choose-infrastructure].  
+
+## Platform applications
+
+Applications can be deployed in any Kubernetes-managed or self-attached clusters, giving you complete flexibility to operate across cloud, on-premise, edge, and air-gapped scenarios. DKP Enterprise users can use the Kommander UI to customize which platform applications to deploy to the cluster in a given workspace. For a list of available platform applications that are included with DKP Enterprise, see [Workspace Platform Applications][workspaceplatform].
```
Comment (cneth) [inline pages/dkp/kommander/2.2/licensing/enterprise/index.md:39]: I don't think that's the correct link.   'Platform applications' are installed by Kommander on the Management cluster.   'Workspace Platform applications' are a subset of the platform applications which can be optionally installed on an attached or managed cluster.   I am not sure where to link for a concise list of the platform applications.

**Review by cneth — CHANGES_REQUESTED** [review]

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md (hunk 2 above)
Comment (mwrightd2) [reply pages/dkp/kommander/2.2/licensing/enterprise/index.md:25]: Sounds good, so per @cmastr recommendation @cneth would a link to currently supported OS table be the solution here?

File: pages/dkp/kommander/2.2/licensing/essential/index.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+excerpt: A section containing all of the features available to users with a DKP Essential license.
+menuWeight: 5
+beta: false
+---
+
+## DKP Essential overview
+
+![DKP Essential Diagram](../img/dkpessentialdiagram.png)
+
+DKP Essential is a FIPS-compliant, self-managed single cluster Kubernetes solution that gives you a feature-rich, easy-to-deploy, and easy-to-manage entry-level cloud container platform. The DKP Essential license gives the user access to the entire Konvoy cluster environment, as well as the Kommander platform application manager for standing up crucial day-2 level production.
```
[the remark this answers is not in the document]
Comment (mwrightd2) [reply pages/dkp/kommander/2.2/licensing/essential/index.md:14]: yeah "standing up crucial day-2 level production" looks like fluff, im just going to cut that off at "...kommander platform application manager."

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md (hunk 7 above)
Comment (mwrightd2) [reply pages/dkp/kommander/2.2/licensing/enterprise/index.md:39]: Just going to delete link sentence. If we dont have supporting platform apps docs besides whats listed then it shouldnt be linked.

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md (hunk 1 above)
Comment (mwrightd2) [reply pages/dkp/kommander/2.2/licensing/enterprise/index.md]: Edited out.

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md (hunk 2 above)
Comment (mwrightd2) [reply pages/dkp/kommander/2.2/licensing/enterprise/index.md]: Removed section and linked to https://docs.d2iq.com/dkp/konvoy/2.1/supported-operating-systems/.

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+## Compatible infrastructure
+
+DKP Enterprise operates across D2iQ's entire range of cloud, on-premise, edge, and air-gapped infrastructures and has support for various OSs, including immutable OSs. See [Supported Operating Systems][supported] for a full list of compatible infrastructure.
+
+For the basics on standing up a DKP Enterprise cluster in one of the listed environments of your choice, see [Choose Infrastructure][choose-infrastructure].  
+
+## Platform applications
+
+Applications can be deployed in any Kubernetes-managed or self-attached clusters, giving you complete flexibility to operate across cloud, on-premise, edge, and air-gapped scenarios. DKP Enterprise users can use the Kommander UI to customize which platform applications to deploy to the cluster in a given workspace.
```
Comment (sonali-mishra-22) [inline pages/dkp/kommander/2.2/licensing/enterprise/index.md:24]: `DKP Enterprise users can use the Kommander UI to customize which platform applications to deploy to the cluster in a given workspace.
`
Let's use DKP UI, as we are using Kommander UI here and DKP UI later in the docs, this leads to inconsistency.

Comment (sonali-mishra-22) [inline pages/dkp/kommander/2.2/licensing/enterprise/index.md:24]: It should be - Applications can be deployed in any DKP managed clusters, giving you complete flexibility to operate across cloud, on-premise, edge, and air-gapped scenarios. DKP Enterprise users can use the Kommander UI to customize which platform applications to deploy to the cluster in a given workspace.

File: pages/dkp/kommander/2.2/licensing/essential/index.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+- **Cert Manager:** A Kubernetes addon to automate the management and issuance of TLS certificates from various issuing sources.
+- **Cluster Autoscaler:** A component that automatically adjusts the size of a Kubernetes cluster so that all pods have a place to run and there are no unneeded nodes.
+
+## Built-in GitOps
+
+DKP Essential comes bundled with GitOps, an operating model for Kubernetes and other cloud native technologies, providing a set of best practices that unify Git deployment, management and monitoring for containerized clusters and applications. GitOps works by using Git as a single source of truth for declarative infrastructure and applications. With GitOps, the use of software agents can alert on any divergence between Git with what is running in a cluster, and if there's a difference, Kubernetes reconcilers automatically update or rollback the cluster depending on the case.
+
+## DKP Essential single cluster UI
+
+Bundled with DKP Essential is a single cluster management UI that can be used in lieu of the bundled CLI. From the UI you can:
```
Comment (sonali-mishra-22) [inline pages/dkp/kommander/2.2/licensing/essential/index.md:48]: Let's make it: 
Bundled with DKP Essential is a single cluster management view of DKP UI that can be used in lieu of the bundled CLI. From the UI you can:

File: pages/dkp/kommander/2.2/licensing/enterprise/index.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+- **Cert Manager:** A Kubernetes addon to automate the management and issuance of TLS certificates from various issuing sources.
+- **Cluster Autoscaler:** A component that automatically adjusts the size of a Kubernetes cluster so that all pods have a place to run and there are no unneeded nodes.
+
+## Built-in GitOps
+
+DKP Enterprise comes bundled with GitOps, an operating model for Kubernetes and other cloud native technologies, providing a set of best practices that unify Git deployment, management and monitoring for containerized clusters and applications. GitOps works by using Git as a single source of truth for declarative infrastructure and applications. With GitOps, the use of software agents can alert on any divergence between Git with what is running in a cluster, and if there's a difference, Kubernetes reconcilers automatically update or rollback the cluster depending on the case.
+
+## DKP Enterprise multi-cluster UI
+
+Bundled with DKP Enterprise is a multi-cluster management UI that can be used in lieu of the bundled CLI. From the UI you can:
```
Comment (sonali-mishra-22) [inline pages/dkp/kommander/2.2/licensing/enterprise/index.md:54]: `Bundled with DKP Enterprise is a multi-cluster management view of DKP UI that can be used in lieu of the bundled CLI. From the UI you can:`

File: pages/dkp/kommander/2.2/licensing/index.md [hunk 12]
```diff
@@ -7,11 +7,11 @@ menuWeight: 299
 beta: false
 ---
 
-The following instructions describe how to download and license Kommander.
+This section contains overviews of the feature sets for DKP Essential and DKP Enterprise licenses, and how to acquire and consume both licenses.
 
 ## Purchase a License
 
-To obtain a valid Kommander license:
+Licenses for both DKP Enterprise and DKP Essential are sold in units of cores. To learn more about both and to obtain a valid license:
```
Comment (msdolbey) [inline pages/dkp/kommander/2.2/licensing/index.md:14]: nit 

```suggestion
Licenses for both DKP Enterprise and DKP Essential are sold in units of cores. To learn more about both, and to obtain a valid license:
```

**Pull request merged**